service/src/main/java/com/ge/predix/acs/encryption/Encryptor.java

```diff
@@ -77,6 +113,6 @@ public void setEncryptionKey(final String encryptionKey) {
         if (null == encryptionKey || encryptionKey.length() != KEY_LENGTH_IN_BYTES) {
             throw new SymmetricKeyValidationException("Encryption key must be string of length " + KEY_LENGTH_IN_BYTES);
         }
-        this.encryptionKey = encryptionKey;
+        this.secretKeySpec = new SecretKeySpec(encryptionKey.getBytes(), ALGO);
```

Comment:
We could unload key length validation to [SecretKeySpec(byte[] key, int offset, int len, String algorithm)](https://docs.oracle.com/javase/8/docs/api/index.html?javax/crypto/spec/SecretKeySpec.html) constructor.  It throws IllegalArgumentException if algorithm is null or key is null, empty, or too short, i.e. key.length-offset<len.